admin/services/access-arrangements.service.js

```diff
@@ -10,4 +16,65 @@ accessArrangementsService.getAccessArrangements = async () => {
   return accessArrangementsDataService.sqlFindAccessArrangements()
 }
 
-module.exports = accessArrangementsService
+/**
+ * Process access arrangements for single pupil
+ * @param {Object} requestData
+ * @param {Number} dfeNumber
+ * @param {Number} userId
+ * @returns {Object}
+ */
+accessArrangementsService.submit = async (requestData, dfeNumber, userId) => {
+  const validationError = accessArrangementsValidator.validate(requestData)
+  if (validationError.hasError()) {
+    throw validationError
+  }
+  const { pupilUrlSlug, accessArrangements: accessArrangementsCodes, questionReaderReason: questionReaderReasonCode } = requestData
+  let pupilAccessArrangements = R.clone(requestData)
+  const accessArrangements = await accessArrangementsDataService.sqlFindAccessArrangementsByCodes(accessArrangementsCodes)
```

Comment:
I don't think there is any plan to support changing access codes during runtime, so this call could be `memoized()`

Reply:
@jon-shipley updated as per our conversation